### DISCUSSION TITLE
Old objects in ioBroker after SW-Update from 3.5 to 3.7

### DISCUSSION THREAD

**Discussion by Hans-JR** [emsesp/EMS-ESP32#2026] [Q&A]
My system (Bosch Condens 5000i and one remote controller CW400) has been running fine for several years now. Today I upgraded the hardware to an LilyGo ESP32 and software from 3.6.0 to 3.7.38. I am using ioBroker with the corrensponding ESP-Adapter (no MQTT). Now I am a little confused about the objects. It seems that many objects are leftover from older versions. Unfortunately I am unsure, which ones I can delete and which ones are the latest (maybe I'm to stupid to understand the changelist). I am attaching a JSON-Export of my EMS-ESP Objects folder. Maybe some is able to let me know, which ones I can delete manually.
Thanks
Hans-Jürgen
[Objects.txt](https://github.com/user-attachments/files/17057380/Objects.txt)

**Comment by MichaelDvP** [comment]
Go to instance->emsesp0->settings->parameter, check the box for "delete states on start" ,
than restart the adapter and all new states will be generated again.

**Reply by Hans-JR** [reply]
Well, I think I will note down all objects that I use for the database and for control and then delete the entire ems-tree. As I mentioned before, this tree contains objects from several releases.....

**Comment by proddy** [comment]
That's a big leap from 3.6.0 to 3.7.0 - 13 months of development, 2200 commits and 711 files changed to be accurate :-)

Unfortunately, there have been breaking changes. We do our best to keep things backwards compatible between releases, and document the changes but some bigger changes do break 3rd party integrations. 

If you want to keep the HA history, there are 3 options
 - don't, and start fresh (see Michael's suggestion)
 - rename the objects in HA, from HA's user interface or via a tool https://emsesp.org/Troubleshooting/#ha-has-messed-up-the-names-of-my-entities
 - change the MQTT "Entity ID format" setting in EMS-ESP to be backward compatible with the old naming. Pick "Single instance, short name (v3.6)". But note, that will still break the change from ww to dhw.

@tp1de just did an excellent write up on https://github.com/emsesp/EMS-ESP32/issues/2001#issuecomment-2360368083

Sorry!
